From ecf1bd6f6bc4e7d20150f828842a40461049d6c6 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Thu, 31 Aug 2017 19:00:47 +0800
Subject: [PATCH 2472/5242] MLK-16348: PxP: fix background issue

commit  6afef859cf67b645ba613c98f0d3cb3657bce1be from
https://source.codeaurora.org/external/imx/linux-imx.git

There was no pxp background register setting, so the
background we see always black.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
(cherry picked from commit d3ee45e2671478854a165f34f91fabac247e8a39)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index c2a0133..7ef4b4c 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -3371,6 +3371,9 @@ static int pxp_2d_op_handler(struct pxps *pxp)
 		break;
 	}
 
+	__raw_writel(proc_data->bgcolor,
+		     pxp->base + HW_PXP_PS_BACKGROUND_0);
+
 	if (proc_data->lut_transform && pxp_is_v3(pxp))
 		set_mux(&path_ctrl0);
 
-- 
1.7.9.5

